reana_commons/snakemake.py

```diff
@@ -42,30 +72,91 @@ def _create_snakemake_workflow(snakefile, configfiles=None):
-        return snakemake_workflow
+        workflow.include(snakefile=snakefile, overwrite_first_rule=True)
+        workflow.check()
+
+        # code copied and adapted from `snakemake.workflow.Workflow.execute()`
+        # in order to build the DAG and calculate the job dependencies.
+        # https://github.com/snakemake/snakemake/blob/75a544ba528b30b43b861abc0ad464db4d6ae16f/snakemake/workflow.py#L525
+        def rules(items):
+            return map(workflow._rules.__getitem__, filter(workflow.is_rule, items),)
+
+        if kwargs.get("keep_target_files"):
+
+            def files(items):
+                return filterfalse(workflow.is_rule, items)
+
+        else:
+
+            def files(items):
+                relpath = (
+                    lambda f: f
+                    if os.path.isabs(f) or f.startswith("root://")
+                    else os.path.relpath(f)
+                )
+                return map(relpath, filterfalse(workflow.is_rule, items))
+
+        if not kwargs.get("targets"):
+            targets = (
+                [workflow.first_rule] if workflow.first_rule is not None else list()
+            )
+
+        prioritytargets = kwargs.get("prioritytargets", [])
+        forcerun = kwargs.get("forcerun", [])
+        until = kwargs.get("until", [])
+        omit_from = kwargs.get("omit_from", [])
+
+        priorityrules = set(rules(prioritytargets))
+        priorityfiles = set(files(prioritytargets))
+        forcerules = set(rules(forcerun))
+        forcefiles = set(files(forcerun))
+        untilrules = set(rules(until))
+        untilfiles = set(files(until))
+        omitrules = set(rules(omit_from))
+        omitfiles = set(files(omit_from))
+
+        targetrules = set(
+            chain(
+                rules(targets),
+                filterfalse(Rule.has_wildcards, priorityrules),
+                filterfalse(Rule.has_wildcards, forcerules),
+                filterfalse(Rule.has_wildcards, untilrules),
+            )
+        )
+        targetfiles = set(chain(files(targets), priorityfiles, forcefiles, untilfiles))
+        dag = DAG(
+            workflow,
+            workflow.rules,
+            targetrules=targetrules,
+            targetfiles=targetfiles,
+            omitfiles=omitfiles,
+            omitrules=omitrules,
+        )
+
+        workflow.persistence = Persistence(dag=dag)
+        dag.init()
+        dag.update_checkpoint_dependencies()
+        dag.check_dynamic()
+        return dag
 
     configfiles = [kwargs.get("input")]
-    valid = snakemake(
-        snakefile=workflow_file,
-        configfiles=[kwargs.get("input")],
-        dryrun=True,
-        quiet=True,
-    )
-    if not valid:
-        raise REANAValidationError("Snakemake specification is invalid.")
+    snakemake_validate(workflow_file=workflow_file, configfiles=configfiles)
+    snakemake_dag = _create_snakemake_dag(workflow_file, configfiles=configfiles)
```

Comment:
*thought:* I just realized it, we do not pass anything to `_create_snakemake_dag` except `workflow_file` and `configfiles`. It basically means that all `kwargs` related checks in `_create_snakemake_dag` don't make sense.

Reply:
Forgot to pass `kwargs` to `_create_snakemake_dag`. I think it's a good practice to pass it as there might be some `kwargs` needed in the future to support more specific Snakemake features beyond MVP.